index.bs

```diff
@@ -4115,6 +4147,810 @@ opened</dfn> steps given |window|, |type| and |message|.
+
+</div>
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.Initiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: js-uint,
+    ?lineNumber: js-uint,
+    ?stackTrace: script.StackTrace,
+    ?request: network.Request
+};
+</pre>
+
+The <code>network.Initiatior</code> type represents the source of a network
+request.
+
+<div algorithm>
+To <dfn>get the initiator</dfn> given |request|:
+
+1. Let |request id| be |request|'s [=request id=].
+
+1. Let |type| be "<code>other</code>".
+
+1. If |request| is a [=CORS-Preflight Request=], set |type| to
+   "<code>preflight</code>".
+
+1. TODO: Get the |type|. It's not quite clear how this ought to work; the CDP
+   data depends on whether the navigation was kicked off by the parser or by
+   script (so e.g. inserting an image from script causes the initiator to be
+   "<code>script</code>"), but that doesn't correspond to anything in Fetch.
+
+1. If |request|'s [=request/initiator type=] is "<code>fetch</code>" or
+   "<code>xmlhttprequest</code>":
+
+  1. Let |stack trace| be the [=current stack trace=].
+
+  1. If |stack trace| has size of 1 or greater, let |line number| be value of the
+     <code>lineNumber</code> field in |stack trace|[0], and let |column number| be
+     the value of the <code>columnNumber</code> field |stack trace|[0]. Otherwise
+     let |line number| and |column number| be 0.
+
+  Otherwise, let |stack trace|, |column number|, and |line number| all be null.
+
+  TODO: Chrome includes the current parser position as column number / line
+  number for parser-inserted resources.
+
+1. Return a [=/map=] matching the <code>network.Initiator</code> production, with
+   the <code>type</code> field set to |type|, the <code>columnNumber</code>
+   field set to |column number| if it's not null, or omitted otherwise, the
+   <code>lineNumber</code> field set to |line number| if it's not null, or
+   omitted otherwise, the <code>stackTrace</code> field set to |stack trace| if
+   it's not null, or omitted otherwise, and the <code>request</code> field set
+   to |request id|.
+
+</div>
+
+#### The network.Request type #### {#type-network-Request}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+network.Request = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a
+string uniquely identifying that request. The identifier for a request resulting from a
+redirect matches that of the request that initiated it.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.RequestData = {
+    request: network.Request,
+    url: text,
+    method: text,
+    headers: [*network.Header],
+    cookies: [*network.Cookie],
+    headersSize: js-uint,
```

Comment:
Do we need the size having the headers themselves? We don't have it in CDP.

Reply:
I believe that HAR wants it, and computing the byte size from the deserialized headers seems complex (you need to re-serialize the headers, and I don't know how that works in the face of header compression).

Reply:
In Chrome the HAR's `headersSize` is always  `-1`, as it [relies](https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/models/har/Log.ts;l=205?q=headersSize&ss=chromium%2Fchromium%2Fsrc:third_party%2Fdevtools-frontend%2Fsrc%2Ffront_end%2F) on [deprecated](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-responseReceivedExtraInfo) field `headersText`.

Reply:
(sorry misclicked on resolve...)

In Firefox DevTools HAR, we do provide the headers size, and yes it would not be trivial to recompute it from deserialized headers. 

And it seems at least one known consumer of HAR is re-implementing this for Chrome to workaround the -1 value: https://github.com/sitespeedio/chrome-har/blob/d23393967e1766fa78c001faec244b6b6378c52d/lib/headers.js#L6-L37

So it seems worth including?

Reply:
I may be wrong here, but isn't `headersSize` used to read the proper data fields from the raw request? And if saw, it is useless in our case, as long as the headers are sent parsed.

Reply:
I don't think headersSize is expected to be used to process data from the request, it's rather a size/performance metric.

If we want to challenge this further we need inputs from consumers for HAR files to know if they could do without the data.